### PR TITLE
backport yt api key patch to tahoe

### PR DESCRIPTION
**Pull request by thraxil**
Same change as #437 but for our Tahoe branch.

### CODE REVIEW

**Comment by johnbaldwin** [comment]
@thraxil I'm reviewing, since these are the same, I'm commenting in the other PR: https://github.com/appsembler/edx-platform/pull/437

File: lms/djangoapps/courseware/views/views.py [hunk 1]
```diff
@@ -228,6 +233,55 @@ def courses(request):
...
+    setting rate limit for  yt_video_metadata API
+    """
+    rate = settings.RATE_LIMIT_FOR_VIDEO_METADATA_API
+
+
+@ensure_csrf_cookie
+@login_required
+@api_view(['GET'])
+@throttle_classes([PerUserVideoMetadataThrottle])
+def yt_video_metadata(request):
```
Comment (johnbaldwin) [inline lms/djangoapps/courseware/views/views.py]: @thraxil @melvinsoft One concern here is that this is not multisite aware. Looks like there a single youtube api key for all users. How are youtube accounts supposed to work on Tahoe for multi tenancy?

**Comment by johnbaldwin** [comment]
@thraxil @melvinsoft Before we push this to Tahoe, are we confident that this addresses multi tenancy for youtube api keys?

File: lms/djangoapps/courseware/views/views.py (hunk 1 above)
Comment (melvinsoft) [reply lms/djangoapps/courseware/views/views.py]: @johnbaldwin So far, all Tahoe sites share the same YT API key. This endpoint is just to fetch and return video metadata.

Comment (johnbaldwin) [reply lms/djangoapps/courseware/views/views.py]: @melvinsoft So no multi-tenancy on YT videos?

**Comment by bryanlandia** [comment]
@melvinsoft I made some changes in the plain Hawthorn version, #437 You might want to take a look

File: lms/djangoapps/courseware/tests/test_video_mongo.py [hunk 2]
```diff
@@ -107,8 +107,7 @@ def test_video_constructor(self):
                 'transcriptLanguages': OrderedDict({'en': 'English', 'uk': u'Українська'}),
                 'ytTestTimeout': 1500,
                 'ytApiUrl': 'https://www.youtube.com/iframe_api',
-                'ytMetadataUrl': 'https://www.googleapis.com/youtube/v3/videos/',
-                'ytKey': None,
+                'lmsRootURL': settings.LMS_ROOT_URL,
```
Comment (melvinsoft) [inline lms/djangoapps/courseware/tests/test_video_mongo.py]: @thraxil What is the use of this variable? is going to be always `tahoe.appsembler.com`. If we want to use the real site name var, we should use something like: `configuration_helpers.get_value('SITE_NAME', settings.SITE_NAME)`

**Comment by melvinsoft** [comment]
@bryanlandia Thanks! Yes, we need to port that change as well. 

@thraxil I added another comment for you, and regarding Bryan's change, do you want me to cherry pick-it here?

**Comment by johnbaldwin** [comment]
@melvinsoft @bryanlandia  @thraxil What do we need to do next to get this merged?

**Comment by thraxil** [comment]
@melvinsoft @bryanlandia are we still working on getting this security patch tested and merged?

**Comment by melvinsoft** [comment]
@thraxil I lost track of it TBH, can you or @bryanlandia summarize what is the remaining work? I'm not super familiar with the issue and I know there is some difference between the Enterprise and Tahoe branches now, should we cherry pick those? I can do it.

**Pull request closed without merging**